tests/nsrt_learning/strips_learning/test_gen_to_spec_learner.py

```diff
@@ -326,3 +325,169 @@ def test_find_unification_and_try_specializing_pnad():
+    machine_type = Type("machine_type", ["on", "configuration", "run"])
+    MachineOn = Predicate("MachineOn", [machine_type],
+                          lambda s, o: s[o[0]][0] > 0.5)
+    NotMachineOn = Predicate("NotMachineOn", [machine_type],
+                             lambda s, o: s[o[0]][0] <= 0.5)
+    MachineConfigured = Predicate("MachineConfigured", [machine_type],
+                                  lambda s, o: s[o[0]][1] > 0.5)
+    MachineRun = Predicate("MachineRun", [machine_type],
+                           lambda s, o: s[o[0]][2] > 0.5)
+    m1 = machine_type("m1")
+    m2 = machine_type("m2")
+    m3 = machine_type("m3")
+
+    # Create states to be used as part of trajectories.
+    all_off_not_configed = State({
+        m1: [0.0, 0.0, 0.0],
+        m2: [0.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0]
+    })
+    m1_off_configed_m2_on = State({
+        m1: [0.0, 1.0, 0.0],
+        m2: [1.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0]
+    })
+    m1_on_configed_m2_on = State({
+        m1: [1.0, 1.0, 0.0],
+        m2: [1.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0]
+    })
+    m1_on_configed_run_m2_on = State({
+        m1: [1.0, 1.0, 1.0],
+        m2: [1.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0]
+    })
+    m3_on = State({
+        m1: [0.0, 0.0, 0.0],
+        m2: [0.0, 0.0, 0.0],
+        m3: [1.0, 0.0, 0.0],
+    })
+    m1_on_m3_on = State({
+        m1: [1.0, 0.0, 0.0],
+        m2: [0.0, 0.0, 0.0],
+        m3: [1.0, 0.0, 0.0],
+    })
+    m1_on_configed = State({
+        m1: [1.0, 1.0, 0.0],
+        m2: [0.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0],
+    })
+    m1_on_configed_run = State({
+        m1: [1.0, 1.0, 1.0],
+        m2: [0.0, 0.0, 0.0],
+        m3: [0.0, 0.0, 0.0],
+    })
+
+    # Create the necessary options and actions.
+    turn_on = SingletonParameterizedOption("TurnOn", lambda s, m, o, p: None)
+    TurnOn = turn_on.ground([], [])
+    turn_on_act = Action([], TurnOn)
+    configure = SingletonParameterizedOption("Configure",
+                                             lambda s, m, o, p: None)
+    Configure = configure.ground([], [])
+    configure_act = Action([], Configure)
+    run = SingletonParameterizedOption("Run", lambda s, m, o, p: None)
+    Run = run.ground([], [])
+    run_act = Action([], Run)
+
+    # Now create the trajectories, goals and tasks.
+    traj1 = LowLevelTrajectory([
+        all_off_not_configed, m1_off_configed_m2_on, m1_on_configed_m2_on,
+        m1_on_configed_run_m2_on
+    ], [configure_act, turn_on_act, run_act], True, 0)
+    traj2 = LowLevelTrajectory(
+        [m3_on, m1_on_m3_on, m1_on_configed, m1_on_configed_run],
+        [turn_on_act, configure_act, run_act], True, 1)
+    goal = {
+        MachineOn([m1]),
+        MachineConfigured([m1]),
+        MachineRun([m1]),
+    }
+    task1 = Task(all_off_not_configed, goal)
+    task2 = Task(m3_on, goal)
+
+    # Define the 2 demos (each with 3 segments) to backchain over.
```

Comment:
instead of defining these segments manually, can we call `segment_trajectory` in segmentation.py on each traj? it should give the same result hopefully...

Reply:
damn - i didn't know this method existed lol - will try!